slack_autoarchive.py

```diff
@@ -78,16 +78,28 @@ def slack_api_http(
                 time.sleep(retry_delay)
 
             if method == 'POST':
-                response = requests.post(uri, data=payload)
+                response = requests.post(uri, json=payload, headers=headers)
             else:
-                response = requests.get(uri, params=payload)
+                response = requests.get(uri, params=payload, headers=headers)
 
             if response.status_code == requests.codes.ok and 'error' in response.json(
-            ) and response.json()['error'] == 'not_authed':
-                self.logger.error(
-                    'Need to setup auth. eg, SLACK_TOKEN=<secret token> python slack-autoarchive.py'
-                )
-                sys.exit(1)
+            ):
+                if response.json()['error'] == 'not_authed':
+                    self.logger.error(
+                        'Need to setup auth. eg, SLACK_TOKEN=<secret token> python slack-autoarchive.py'
+                    )
+                    sys.exit(1)
+                elif response.json()['error'] == 'not_in_channel':
+                    self.logger.error(
+                        'Need to add bot to channel described by ' + str(payload)
+                    )
+                    self.join_channel(payload['channel'])
+                    return response.json()
```

Comment:
```suggestion
                    return self.slack_api_http(api_endpoint, payload, method)
```

TL;DR: Without this fix, the bot will archive **EVERY** channel

If you just return the `response` the following will happen:
1. Bot tries to read history -> rejected
2. Bot joins channel
3. Return `repsonse`, but in `response` is the message "not_in_channel"
4. Because no messages are in the request -> bot will archive the channel

Reply:
@nilsreichardt : Thanks for catching the bug. Suggested changes applied.